cpp/cmake/Modules/SetGPUArchs.cmake

```diff
@@ -35,27 +35,17 @@ if(CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 9)
   list(REMOVE_ITEM SUPPORTED_CUDA_ARCHITECTURES "70")
 endif()
 
-if(${PROJECT_NAME}_BUILD_FOR_ALL_ARCHS)
+if(CUDF_BUILD_FOR_ALL_ARCHS)
   set(CMAKE_CUDA_ARCHITECTURES ${SUPPORTED_CUDA_ARCHITECTURES})
-elseif(${PROJECT_NAME}_BUILD_FOR_DETECTED_ARCHS)
-  include(${PROJECT_SOURCE_DIR}/cmake/EvalGpuArchs.cmake)
+elseif(CUDF_BUILD_FOR_DETECTED_ARCHS)
+  include(${CUDF_SOURCE_DIR}/cmake/Modules/EvalGPUArchs.cmake)
   evaluate_gpu_archs(CMAKE_CUDA_ARCHITECTURES)
 endif()
 
-if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.18)
-  # CMake architecture list entry of "80" means to build compute and sm.
-  # What we want is for the newest arch only to build that way
-  # while the rest built only for sm.
-  list(SORT CMAKE_CUDA_ARCHITECTURES ORDER ASCENDING)
-  list(POP_BACK CMAKE_CUDA_ARCHITECTURES latest_arch)
-  list(TRANSFORM CMAKE_CUDA_ARCHITECTURES APPEND "-real")
-  list(APPEND CMAKE_CUDA_ARCHITECTURES ${latest_arch})
-else()
-  foreach(arch IN LISTS CMAKE_CUDA_ARCHITECTURES)
-    string(APPEND CMAKE_CUDA_FLAGS " -gencode=arch=compute_${arch},code=sm_${arch}")
-  endforeach()
-
-  list(GET CMAKE_CUDA_ARCHITECTURES -1 ptx)
-  string(APPEND CMAKE_CUDA_FLAGS " -gencode=arch=compute_${ptx},code=compute_${ptx}")
-  unset(CMAKE_CUDA_ARCHITECTURES)
-endif()
+# CMake architecture list entry of "80" means to build compute and sm.
```

Comment:
We should update this to be:

```cmake
if(${PROJECT_NAME}_BUILD_FOR_ALL_ARCHS)
  # CMake architecture list entry of "80" means to build compute and sm. What we want is for the
  # newest arch only to build that way while the rest built only for sm.
  list(POP_BACK CMAKE_CUDA_ARCHITECTURES latest_arch)
  list(TRANSFORM CMAKE_CUDA_ARCHITECTURES APPEND "-real")
  list(APPEND CMAKE_CUDA_ARCHITECTURES ${latest_arch})
elseif(${PROJECT_NAME}_BUILD_FOR_DETECTED_ARCHS)
  list(TRANSFORM CMAKE_CUDA_ARCHITECTURES APPEND "-real")
endif()
```

This would make sure when building for a developers local machine we only build for `sm` instead of both `compute` and `sm`